### PR TITLE
Fix id reference for generate commands

### PR DESCRIPTION
**Pull request by KohHanMing**
Fixed the id references in generatePeople and generateLocations such that they always refer to the personBook so the correct person is referenced regardless of the state of the filtered person list

### CODE REVIEW

**Review by hopinxian — APPROVED** [review]
LGTM

File: src/test/java/seedu/address/logic/commands/GeneratePeopleCommandTest.java [hunk 1]
```diff
@@ -120,4 +122,31 @@ public void execute_validInput_success() {
...
+    public void execute_invalidInputFromViewingAllQuarantined_throwCommandException() {
+        String expectedMessage = MESSAGE_PERSON_IS_NOT_INFECTED;
+        Model modelForAllInfected = model;
+        modelForAllInfected.updateFilteredPersonList(PREDICATE_SHOW_ALL_QUARANTINED);
+        Index index = Index.fromOneBased(1);
+        GeneratePeopleCommand command = new GeneratePeopleCommand(index);
+        try {
+            command.execute(modelForAllInfected);
+        } catch (CommandException e) {
+            assertTrue(e.getMessage().equals(expectedMessage));
```
Comment (hopinxian) [inline src/test/java/seedu/address/logic/commands/GeneratePeopleCommandTest.java:149]: Suggest to replace with the assertThrow from Command utils

File: src/test/java/seedu/address/logic/commands/GenerateLocationsCommandTest.java [hunk 2]
```diff
@@ -105,4 +107,32 @@ public void execute_validInput_success() {
...
+    public void execute_invalidInputFromViewingAllQuarantined_throwCommandException() {
+        String expectedMessage = MESSAGE_PERSON_IS_NOT_INFECTED;
+        Model modelForAllQuarantined = model;
+        modelForAllQuarantined.updateFilteredPersonList(PREDICATE_SHOW_ALL_QUARANTINED);
+        Index index = Index.fromOneBased(1);
+        GenerateLocationsCommand command = new GenerateLocationsCommand(index);
+        try {
+            command.execute(modelForAllQuarantined);
+        } catch (CommandException e) {
+            assertTrue(e.getMessage().equals(expectedMessage));
```
Comment (hopinxian) [inline src/test/java/seedu/address/logic/commands/GenerateLocationsCommandTest.java:135]: Suggest to replace with AssertThrow from command utils

**Review by hopinxian — COMMENTED** [review]
some slight improvements to use existing methods

**Review by siangernlow — APPROVED** [review]
LGTM

**Review by siangernlow — CHANGES_REQUESTED** [review]
Failing tests have to be fixed

File: src/test/java/seedu/address/logic/commands/GeneratePeopleCommandTest.java (hunk 1 above)
Comment (KohHanMing) [reply src/test/java/seedu/address/logic/commands/GeneratePeopleCommandTest.java:184]: Done!

File: src/test/java/seedu/address/logic/commands/GenerateLocationsCommandTest.java (hunk 2 above)
Comment (KohHanMing) [reply src/test/java/seedu/address/logic/commands/GenerateLocationsCommandTest.java:162]: Done!

File: src/test/java/seedu/address/logic/commands/GenerateLocationsCommandTest.java [hunk 3]
```diff
@@ -107,4 +110,56 @@ public void execute_validInput_success() {
...
```
Comment (hopinxian) [inline src/test/java/seedu/address/logic/commands/GenerateLocationsCommandTest.java]: This test is repeated and ought to be deleted.

File: src/test/java/seedu/address/logic/commands/GeneratePeopleCommandTest.java [hunk 4]
```diff
@@ -113,16 +118,70 @@ public void execute_noPeopleFound_throwCommandException() {
...
+        modelForAllInfected.updateFilteredPersonList(PREDICATE_SHOW_ALL_INFECTED);
+        Index index = Index.fromOneBased(1);
+        GeneratePeopleCommand command = new GeneratePeopleCommand(index);
+        assertThrows(CommandException.class, () -> command.execute(model));
+        try {
+            command.execute(model);
+        } catch (CommandException e) {
+            assertTrue(e.getMessage().equals(expectedMessage));
+        }
+    }
```
Comment (hopinxian) [inline src/test/java/seedu/address/logic/commands/GeneratePeopleCommandTest.java:150]: Repeated tests are detected here.

File: src/test/java/seedu/address/logic/commands/GeneratePeopleCommandTest.java [hunk 5]
```diff
@@ -113,16 +118,70 @@ public void execute_noPeopleFound_throwCommandException() {
...
+    public void execute_invalidInputFromViewingAllQuarantined_throwCommandException() {
+        String expectedMessage = MESSAGE_PERSON_IS_NOT_INFECTED;
+        Model modelForAllInfected = model;
+        modelForAllInfected.updateFilteredPersonList(PREDICATE_SHOW_ALL_QUARANTINED);
+        Index index = Index.fromOneBased(1);
+        GeneratePeopleCommand command = new GeneratePeopleCommand(index);
+        try {
+            command.execute(modelForAllInfected);
+        } catch (CommandException e) {
+            assertTrue(e.getMessage().equals(expectedMessage));
```
Comment (hopinxian) [inline src/test/java/seedu/address/logic/commands/GeneratePeopleCommandTest.java]: Repeated tests are detected here.

**Review by hopinxian — CHANGES_REQUESTED** [review]
After removing the repeated tests, LGTM

**Review by siangernlow — APPROVED** [review]

File: src/test/java/seedu/address/logic/commands/GeneratePeopleCommandTest.java (hunk 5 above)
Comment (KohHanMing) [reply src/test/java/seedu/address/logic/commands/GeneratePeopleCommandTest.java]: Remoed

File: src/test/java/seedu/address/logic/commands/GeneratePeopleCommandTest.java (hunk 4 above)
Comment (KohHanMing) [reply src/test/java/seedu/address/logic/commands/GeneratePeopleCommandTest.java:150]: Removed

File: src/test/java/seedu/address/logic/commands/GenerateLocationsCommandTest.java (hunk 3 above)
Comment (KohHanMing) [reply src/test/java/seedu/address/logic/commands/GenerateLocationsCommandTest.java]: Removed

**Review by hopinxian — APPROVED** [review]
LGTM

**Pull request merged**